src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/3.sql

```diff
@@ -82,6 +82,31 @@ BEGIN
 END
 GO
 
+--
+--  STORED PROCEDURE
+--      SelectMaxSupportedSchemaVersion
+--
+--  DESCRIPTION
+--      Selects the highest supported completed schema version
+--  PARAMETERS
+--      @version
+--          * The version number
+--
+--  RETURNS
+--      The highest version
+--
+CREATE PROCEDURE dbo.SelectMaxSupportedSchemaVersion
+    @version int
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT MAX(Version)
+    FROM SchemaVersion
+    WHERE Status = 'complete' AND Version <= @version
```

Comment:
This just seems like it's basically giving the current version in most cases unless you happen to hit an instance that hasn't updated to the latest version.

Reply:
Updated the query, please review and let me know your thoughts5